instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py

```diff
@@ -0,0 +1,127 @@
+async_produce_hook (Callable) - a function with extra user-defined logic to be performed before sending the message
+this function signature is:
+def async_produce_hook(span: Span, args, kwargs)
+async_consume_hook (Callable) - a function with extra user-defined logic to be performed after consuming a message
+this function signature is:
+def async_consume_hook(span: Span, record: kafka.record.ABCRecord, args, kwargs)
+for example:
+
+.. code: python
+    from opentelemetry.instrumentation.kafka import AIOKafkaInstrumentor
+    from aiokafka import AIOKafkaProducer, AIOKafkaConsumer
+
+    async def async_produce_hook(span, args, kwargs):
+        if span and span.is_recording():
+            span.set_attribute("custom_user_attribute_from_async_response_hook", "some-value")
+    async def async_consume_hook(span, record, args, kwargs):
+        if span and span.is_recording():
+            span.set_attribute("custom_user_attribute_from_consume_hook", "some-value")
+
+    # instrument kafka with produce and consume hooks
+    AIOKafkaInstrumentor().instrument(async_produce_hook=async_produce_hook, async_consume_hook=async_consume_hook)
+
+    # Using kafka as normal now will automatically generate spans,
+    # including user custom attributes added from the hooks
+    producer = AIOKafkaProducer(bootstrap_servers=['localhost:9092'])
+    await producer.send('my-topic', b'raw_bytes')
+
+API
+___
+"""
+from typing import Collection
+
+import aiokafka
+from wrapt import wrap_function_wrapper
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.aiokafka.package import _instruments
+from opentelemetry.instrumentation.aiokafka.utils import (
+    _wrap_anext,
+    _wrap_send,
+)
+from opentelemetry.instrumentation.aiokafka.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import unwrap
+
+
+class AIOKafkaInstrumentor(BaseInstrumentor):
+    """An instrumentor for kafka module
+    See `BaseInstrumentor`
+    """
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        """Instruments the kafka module
+
+        Args:
+            **kwargs: Optional arguments
+                ``tracer_provider``: a TracerProvider, defaults to global.
+                ``async_produce_hook``: a callable to be executed just before producing a message
+                ``async_consume_hook``: a callable to be executed just after consuming a message
+        """
+        tracer_provider = kwargs.get("tracer_provider")
+        async_produce_hook = kwargs.get("async_produce_hook")
+        async_consume_hook = kwargs.get("async_consume_hook")
+
+        tracer = trace.get_tracer(
+            __name__,
+            __version__,
+            tracer_provider=tracer_provider,
+            schema_url="https://opentelemetry.io/schemas/1.11.0",
```

Comment:
Question: 
Since it's new instrumentation, what do you think of using the [stable](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0) semantic conventions?

Reply:
kafka and messaging semconv are still experimental

Reply:
Shouldn't we use latest semconv instead of an old version?

Reply:
Yes! We discussed this at SIG and makes sense to use the latest semconv for span attributes and also use the symbols from `_incubating`

Reply:
@dimastbk are you able to work on this?

Reply:
Yes